plugins/bigquery/dbt/adapters/bigquery/impl.py

```diff
@@ -346,6 +346,36 @@ def execute_model(self, model, materialization, sql_override=None,
 
         return res
 
+    @available.parse_none
+    def is_replaceable(self, relation, conf_partition, conf_cluster):
+        """
+        Check if a given partition and clustering column spec for a table
+        can replace an existing relation in the database. BigQuery does not
+        allow tables to be replaced with another table that has a different
+        partitioning spec. This method returns True if the given config spec is
+        identical to that of the existing table.
+        """
+        try:
+            table = self.connections.get_bq_table(
+                database=relation.database,
+                schema=relation.schema,
+                identifier=relation.identifier
+            )
+        except google.cloud.exceptions.NotFound:
+            return True
+
+        table_partition = table.time_partitioning
+        if table_partition is not None:
+            table_partition = table_partition.field
+
+        table_cluster = table.clustering_fields
+
+        if isinstance(conf_cluster, str):
+            conf_cluster = [conf_cluster]
+
+        return table_partition == conf_partition \
+            and table_cluster == conf_cluster
```

Comment:
Does order matter? If not, we should compare `set(table_cluster) == set(conf_cluster)`.

Reply:
Yeah -- the order is significant -- clustering works like ordering whereby the table is clustered by the first clustering key, then the second, and so on.

This query fails if you run it twice, swapping the order of the clustering keys on the second run:
```
create or replace table dbt_dbanin.debug_clustering
partition by date_day
cluster by id, name
as (
  select current_date as date_day, 1 as id, 'drew' as name
);
```